File: src/api/consumer/media.rs
use axum::routing::{get, post};
use crate::api::web_path::{WebPath, WebPathType};
use crate::application::consumer::media_service;
use crate::common::error::Error;
use crate::common::result::ApiResponse;
use crate::model::prelude::ListData;
use crate::domain::args::a_media::*;
use crate::domain::model::m_media::*;
use axum::{extract::Query, Json};
use validator::Validate;

pub async fn generate_upload_url(
    Json(args): Json<GenerateUploadUrlArgs>,
) -> Result<Json<ApiResponse<UploadUrlResp>>, Error> {
    args.validate()?;

    let result = media_service::generate_upload_url(GenerateUploadUrlParams {
        file_name: args.file_name,
        file_type: match args.file_type.as_str() {
            "video" => FileType::Video,
            _ => FileType::Image,
        },
        consumer_id: args.consumer_id,
    })
    .await?;

    Ok(Json(ApiResponse::success(result)))
}

pub async fn confirm_upload(
    Json(args): Json<ConfirmUploadArgs>,
) -> Result<Json<ApiResponse<MediaFileModel>>, Error> {
    args.validate()?;

    let result = media_service::confirm_upload(ConfirmUploadParams {
        oss_key: args.oss_key,
        file_name: args.file_name,
        file_type: match args.file_type.as_str() {
            "video" => FileType::Video,
            _ => FileType::Image,
        },
        file_size: args.file_size,
        consumer_id: args.consumer_id,
        mime_type: None,
    })
    .await?;

    Ok(Json(ApiResponse::success(result)))
}

pub async fn list_files(
    Query(params): Query<MediaListArgs>,
) -> Result<Json<ApiResponse<ListData<MediaFileModel>>>, Error> {
    let (items, total) = media_service::list_files(
        MediaFileListParams {
            consumer_id: params.consumer_id,
            file_type: params.file_type,
            file_name: None,
            start_time: params.start_time,
            end_time: params.end_time,
        },
    )
    .await?;

    let page_size = params.page_size.unwrap_or(10) as u64;
    let page_num = params.page_num.unwrap_or(1) as u64;

    Ok(Json(ApiResponse::success(ListData {
        list: items,
        total,
        total_pages: (total + page_size - 1) / page_size,
        page_num,
    })))
}

pub fn media_api() -> WebPath {
    WebPath::new()
        .route("/upload-url", WebPathType::Post, Some("生成上传URL"), post(generate_upload_url))
        .route("/confirm", WebPathType::Post, Some("确认上传"), post(confirm_upload))
        .route("/list", WebPathType::Get, Some("媒体文件列表"), get(list_files))
}

File: frontend/src/api/modules/consumer/media.ts
import axios from "axios";
import { Message } from "@arco-design/web-vue";

export interface MediaFile {
  id: number;
  consumer_id: number;
  file_name: string;
  file_type: "image" | "video";
  file_format?: string;
  file_size: number;
  file_url?: string;
  thumbnail_url?: string;
  oss_bucket?: string;
  oss_key?: string;
  is_deleted: boolean;
  created_at?: string;
  deleted_at?: string;
}

export interface UploadUrlResp {
  oss_key: string;
  upload_url: string;
  expires_in: number;
}

export interface GenerateUploadUrlParams {
  consumer_id: number;
  file_name: string;
  file_type: string;
}

export interface ConfirmUploadParams {
  consumer_id: number;
  oss_key: string;
  file_name: string;
  file_type: string;
  file_size: number;
  thumbnail_key?: string;
}

export interface PageParams {
  page_num?: number;
  page_size?: number;
  consumer_id?: number;
  file_type?: string;
  start_time?: string;
  end_time?: string;
}

export const mediaApi = {
  generateUploadUrl: async (params: GenerateUploadUrlParams) => {
    const res = await axios.post("/media/upload-url", params);
    return res.data;
  },

  confirmUpload: async (params: ConfirmUploadParams) => {
    const res = await axios.post("/media/confirm", params);
    if (res.data.message !== "success") {
      Message.error(res.data.message);
    }
    return res.data;
  },

  list: async (params: PageParams) => {
    const res = await axios.get("/media/list", { params });
    return res.data;
  },

  delete: async (id: number) => {
    const res = await axios.delete(`/media/${id}`);
    if (res.data.message !== "success") {
      Message.error(res.data.message);
    }
    return res.data;
  }
};
